src/components/Dropdown/Dropdown.tsx

```diff
@@ -96,42 +125,79 @@ export class Dropdown extends Component<DropdownProps> {
 
   render() {
     const {
+      id: propId,
       children,
-      name,
+      labelProps,
+      labelText,
+      labelTextProps,
+      labelMode = 'top',
+      'aria-labelledby': ariaLabelledBy,
+      visualPlaceholder: name,
       className,
       dropdownButtonProps = {},
       dropdownPopoverProps = {},
       menuPopoverComponent: MenuPopoverComponentReplace,
       dropdownItemProps = {},
-      changeNameToSelection = true,
+      changeVisualPlaceholderToSelection: changeNameToSelection = true,
       ...passProps
     } = this.props;
-
     if (React.Children.count(children) < 1) {
       logger.warn(`Dropdown '${name}' does not contain items`);
       return null;
     }
 
+    const id = idGenerator(propId);
+    const labelId = `${id}-label`;
+    const buttonId = !!dropdownButtonProps.id
+      ? dropdownButtonProps.id
+      : `${id}_button`;
+
     const { selectedName } = this.state;
     const passDropdownButtonProps = {
       ...dropdownButtonProps,
-      className: classnames(buttonClassName, dropdownButtonProps.className),
+      id: buttonId,
+      className: classnames(
+        dropdownClassNames.button,
+        dropdownButtonProps.className,
+      ),
+      'aria-labelledby': `${
+        !!ariaLabelledBy ? `${ariaLabelledBy} ` : ''
+      }${labelId}`,
     };
```

Comment:
I'd prefer to do this above with the id logic to keep the props clean and readable.

Reply:
Fixed. Added a couple of empty lines to improve readability also.